src/sentry/static/sentry/app/components/smartSearchBar/index.tsx

```diff
@@ -352,19 +447,20 @@ class SmartSearchBar extends React.Component {
     if (key === 'ArrowDown' || key === 'ArrowUp') {
       evt.preventDefault();
 
-      const {searchItems, flatSearchItems, activeSearchItem} = this.state;
+      const {flatSearchItems, activeSearchItem} = this.state;
+      const searchGroups = [...this.state.searchGroups];
+
       const [groupIndex, childrenIndex] = isSelectingDropdownItems
-        ? findSearchItemByIndex(searchItems, activeSearchItem)
+        ? filterSearchGroupsByIndex(searchGroups, activeSearchItem)
         : [];
 
       // Remove the previous 'active' property
       if (typeof groupIndex !== 'undefined') {
         if (
-          searchItems[groupIndex] &&
-          searchItems[groupIndex].children &&
-          searchItems[groupIndex].children[childrenIndex]
+          childrenIndex !== undefined &&
+          searchGroups[groupIndex]?.children?.[childrenIndex]
         ) {
-          delete searchItems[groupIndex].children[childrenIndex].active;
+          delete searchGroups[groupIndex].children[childrenIndex].active;
```

Comment:
We actually need to copy things deeper here, since we're actually mutating state.. we don't see the side affect of this because we call `setState` causing a re-render almost immediately after.